src/jbi/configuration.py

```diff
@@ -1,83 +1,33 @@
-"""Parsing and validating the YAML configuration occurs within this module"""
-import importlib
+"""
+Parsing and validating the YAML configuration occurs within this module
+"""
 import logging
-from inspect import signature
-from types import ModuleType
-from typing import Any, Dict, Optional
+from functools import lru_cache
 
-import yaml
-from yaml import Loader
+from pydantic import ValidationError
 
 from src.app import environment
+from src.jbi.model import Actions
 
 settings = environment.get_settings()
 jbi_logger = logging.getLogger("src.jbi")
 
 
 class ConfigError(Exception):
-    """Errors used to determine when the Configuration is invalid"""
-
-
-class ProcessError(Exception):
-    """Error when an exception occurs during processing"""
+    """Error when an exception occurs during processing config"""
 
 
+@lru_cache
 def get_yaml_configurations(
```

Comment:
I think it would be clearer if we would use the term *actions configuration* throughout the whole code base, instead of *yaml configuration*.

For example, this function could be called `get_actions()`, especially since it's in the configuration module. 

`actions = configuration.get_actions()`